clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -89,18 +89,26 @@ public void append(long offset, Record record) {
     /**
      * Append a new record and offset to the buffer
      */
-    public void append(long offset, long timestamp, byte[] key, byte[] value) {
+    public void append(long offset, long timestamp, long checksum, byte[] key, byte[] value) {
         if (!writable)
             throw new IllegalStateException("Memory records is not writable");
 
         int size = Record.recordSize(key, value);
         compressor.putLong(offset);
         compressor.putInt(size);
-        compressor.putRecord(timestamp, key, value);
+        compressor.putRecord(timestamp, checksum, key, value);
         compressor.recordWritten(size + Records.LOG_OVERHEAD);
     }
 
     /**
+     * Append a new record and offset to the buffer
+     */
+    public void append(long offset, long timestamp, byte[] key, byte[] value) {
+        long crc = Record.computeChecksum(timestamp, key, value, CompressionType.NONE, 0, -1);
```

Comment:
Seems like we want to save the `CompressionType` passed in the constructor and then use it here, no?